lib/views/chat_page.dart

```diff
@@ -1,17 +1,66 @@
 import 'package:flutter/material.dart';
+import 'package:pallinet/components/appointment_card.dart';
+import 'package:pallinet/models/session_manager.dart';
+import 'package:pallinet/firestore/firestore.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:pallinet/utils.dart';
 
-class ChatPage extends StatelessWidget {
+class ChatPage extends StatefulWidget {
   const ChatPage({super.key});
 
+  @override
+  State<ChatPage> createState() => _ChatPageState();
+}
+
+class _ChatPageState extends State<ChatPage> {
+  late final SessionManager _prefs;
+
+  @override
+  void initState() {
+    _prefs = SessionManager();
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+  }
+
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-        body: Center(
-      child: Column(
-        children: const [
-          Text("ChatPage"),
-        ],
+    return Column(children: [
+      Expanded(
+        flex: 8,
+        child: FutureBuilder<List<dynamic>>(
+          future:
+              _prefs.getUid().then((uid) => retrieveAppointmentsPatients(uid)),
```

Comment:
I presume this will be changed, but rn, if I try to click into a chat from Physician, it'll throw a null error.

Reply:
This will be fixed to the appropriate method on another pull request